### PR TITLE
Create Budget Posts

### LINKED ISSUE #1: "Create Budget" page
This should let you create a budget (although a budget is just a name so far).

This should be similar to the "create book" stuff.

### PR DESCRIPTION
**Pull request by mso122591**
Allows users to create budgets and displays them in the menu.

Fixes #1

### CODE REVIEW

File: views/budget_detail.pug [hunk 1]
```diff
@@ -0,0 +1,15 @@
+extends layout
+
+block content
+
+  h1 Budget: #{budget.name}
+  p Income: #{budget.income}
+  p Expeneses: #{budget.expenses}
+  - var total = budget.income - budget.expenses
```
Comment (glasserc) [inline views/budget_detail.pug]: This could be a virtual property on Budget.

File: views/layout.pug [hunk 2]
```diff
@@ -25,7 +25,10 @@ html(lang='en')
               li 
                 a(href='/catalog/bookinstances') All book-instances
               li 
+               a(href='/catalog/budgets') All budgets
```
Comment (glasserc) [inline views/layout.pug]: The indentation here is a bit weird -- just one space. (Everything else is two spaces.)

File: routes/users.js [hunk 3]
```diff
@@ -6,4 +6,11 @@ router.get('/', function(req, res, next) {
   res.send('respond with a resource');
 });
 
+//test you are so cool
+router.get('/cool', function(req, res) {
+ 
+  res.send('you are so cool');
+ 
+});
```
Comment (glasserc) [inline routes/users.js]: We should get rid of this.

File: views/budget_list.pug [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  ul
+  each budget in budget_list
+    - var total = budget.income - budget.expenses
+    li 
+      a(href=budget.url) #{budget.name} 
+      p Total Funds Available: #{total}
+      
+
+  else
+    li There are no budgets.
```
Comment (glasserc) [inline views/budget_list.pug]: Can you maybe check your text editor settings to see if there's a way to make sure that there's a newline at end of file?

File: views/budget_detail.pug (hunk 1 above)
Comment (rlaw125) [reply views/budget_detail.pug]: Looks okay to me.

File: views/budget_list.pug (hunk 4 above)
Comment (mso122591) [reply views/budget_list.pug]: Yea I'll take a look

**Pull request merged**